lambdas/cfn-oidc-identity-provider/template.yaml

```diff
@@ -0,0 +1,67 @@
+AWSTemplateFormatVersion: 2010-09-09
+Transform: AWS::Serverless-2016-10-31
+Description: OIDC identity provider
+Metadata:
+  Source: https://github.com/mozilla/security/tree/master/operations/cloudformation-templates/oidc_identity_provider/
+Parameters:
+  Url:
+    Type: String
+    Description: The URL of the identity provider. The URL must begin with https:// and should correspond to the iss claim in the provider's OpenID Connect ID tokens
+    AllowedPattern: https://.*
+    ConstraintDescription: The URL must begin with https://
+  ClientIDList:
+    Type: String
+    Description: A comma delimited list of client IDs (also known as audiences)
+  ThumbprintList:
+    Type: String
+    Description: A comma delimited list list of server certificate thumbprints for the OpenID Connect (OIDC) identity provider's server certificates (also known as CA Thumbprints)
+Resources:
+  IdentityProvider:
+    Type: Custom::IdentityProvider
+    Properties:
+      ServiceToken: !GetAtt IdentityProviderCreatorFunction.Arn
```

Comment:
Ok, I think I'm following this ok, but haven't made a custom resource yet. From what I'm reading all it really takes is to use `Custom::<FILL_IN_THE_BLANK>` as your type but the real business is in specifying this `ServiceToken`. So that makes sense, in that it sort of mirrors how a Macro works, in both cases you're just handing off the business to a `AWS::Serverless::Function`.